charmcraft/reactive_plugin.py

```diff
@@ -179,17 +200,24 @@ def build(
     cmd.extend(["-o", build_dir])
 
     try:
-        subprocess.run(cmd, check=True)
+        run_charm_tool(cmd)
     except subprocess.CalledProcessError as call_error:
-        if call_error.returncode >= 200:
-            return call_error.returncode
+        return call_error.returncode
     finally:
         charm_build_dir.unlink()
 
     return 0
 
 
 if __name__ == "__main__":
+    logpath = get_managed_environment_log_path()
+    emit.init(
```

Comment:
Do not use Emitter from the plugin's subprocess. We actually went away from that model in `charm_builder.py` a few days ago, towards a more generic "plugin subprocess" model as always have been used by Snapcraft.

Just `print()` all the progress/debug info that you need, it will be handled properly.

Reply:
Ah, that makes it much simpler, thank you for sharing that development. :+1: